### PR TITLE
Include process args in `*.fuzzil.history` when `inspect` and `argumentRandomization` options are supplied

### PR DESCRIPTION
**Pull request by 0xedward**
This PR writes the process arguments to `*.fuzzil.history` files when `--inspect` and `--argumentRandomization` options are supplied. Having the process arguments per testcase is helpful when reviewing coverage, reusing the corpus generated by Fuzzilli with other fuzzers, and reproducing of interesting files.

### Test Plan
1. Added a print statement to print out the program UUIDs before execution
2. Run Fuzzilli
    - `swift run FuzzilliCli --profile=hermes ../fuzz-build/bin/hermes --argumentRandomization --storagePath=./corpus --inspect`
3. Stopped running Fuzzilli
4. Resume Fuzzilli to have a different set of arguments logged to `*.fuzzil.history` files
    - `swift run FuzzilliCli --profile=hermes ../fuzz-build/bin/hermes --argumentRandomization --storagePath=./corpus --inspect --resume --staticCorpus`
5. Checked that process arguments are logged for each program UUID printed out
7. Deleted the corpus directory
8. Checked that Fuzzilli doesn't write process arguments to `*.fuzzil.history` when `argumentRandomization` is not supplied by running Fuzzilli (`swift run FuzzilliCli --profile=hermes ../fuzz-build/bin/hermes --storagePath=./corpus --inspect`)

### Example `.fuzzil.history` file
```
// ===== [ Program 2E0482D9-6787-442C-8305-FEDDC5DBD871 ] =====
// Start of prefix code
// Executing code generator ObjectConstructorGenerator
v0 <- BeginConstructor -> v1
    v2 <- LoadInteger '1073741825'
    SetProperty v1, 'e', v2
    SetProperty v1, 'h', v2
EndConstructor
v3 <- Construct v0, []
v4 <- Construct v0, []
v5 <- Construct v0, []
// Code generator finished
// Executing code generator BigIntGenerator
v6 <- LoadBigInt '-256'
v7 <- LoadBigInt '6'
v8 <- LoadBigInt '127'
// Code generator finished
// End of prefix code. 7 variables are now visible
v9 <- LoadString 'o'
v10 <- LoadString 'symbol'
v11 <- LoadInteger '9007199254740992'
v12 <- LoadInteger '-9007199254740991'
v13 <- CallMethod (guarded) v10, 'includes', []
v14 <- Compare v9, '>=', v13
BeginIf v14
    v15 <- CallFunction (guarded) v13, [v13, v13, v12, v11]
EndIf
v16 <- LoadFloat '2.2250738585072014e-308'
v17 <- LoadString 'o'
v18 <- LoadString 'symbol'
v19 <- CallMethod v18, 'includes', []
v20 <- Compare v17, '>=', v18
v21 <- LoadFloat '101.4735377839952'


// ===== [ Program 619119C4-4149-4F7E-8F15-F0A0D519BDD6 ] =====
// Mutating 2E0482D9-6787-442C-8305-FEDDC5DBD871 with OperationMutator
// ARGS: ../fuzz-build/bin/hermes --reprl --lazy-compilation
v0 <- BeginConstructor -> v1
    // Mutating next operation
    v2 <- LoadInteger '1770202423'
    SetProperty v1, 'e', v2
    SetProperty v1, 'h', v2
EndConstructor
v3 <- Construct v0, []
v4 <- Construct v0, []
v5 <- Construct v0, []
v6 <- LoadBigInt '-256'
v7 <- LoadBigInt '6'
// Mutating next operation
v8 <- LoadBigInt '4'
v9 <- LoadString 'o'
// Mutating next operation
v10 <- LoadString 'undefined'
v11 <- LoadInteger '9007199254740992'
v12 <- LoadInteger '-9007199254740991'
v13 <- CallMethod (guarded) v10, 'includes', []
v14 <- Compare v9, '>=', v13
BeginIf v14
    v15 <- CallFunction (guarded) v13, [v13, v13, v12, v11]
EndIf
v16 <- LoadFloat '2.2250738585072014e-308'
v17 <- LoadString 'o'
// Mutating next operation
v18 <- LoadString '9223372036854775807'
v19 <- CallMethod v18, 'includes', []
v20 <- Compare v17, '>=', v18
v21 <- LoadFloat '101.4735377839952'
// Program may be interesting due to new coverage: 219 newly discovered edges in the CFG of the target


// ===== [ Program 28AFDA08-4BBB-4ACB-96B0-11D44389B683 ] =====
// Mutating 619119C4-4149-4F7E-8F15-F0A0D519BDD6 with CodeGenMutator
// ARGS: ../fuzz-build/bin/hermes --reprl --lazy-compilation
v0 <- BeginConstructor -> v1
    v2 <- LoadInteger '1770202423'
    SetProperty v1, 'e', v2
    SetProperty v1, 'h', v2
EndConstructor
v3 <- Construct v0, []
v4 <- Construct v0, []
v5 <- Construct v0, []
v6 <- LoadBigInt '-256'
v7 <- LoadBigInt '6'
v8 <- LoadBigInt '4'
v9 <- LoadString 'o'
v10 <- LoadString 'undefined'
v11 <- LoadInteger '9007199254740992'
v12 <- LoadInteger '-9007199254740991'
v13 <- CallMethod (guarded) v10, 'includes', []
v14 <- Compare v9, '>=', v13
BeginIf v14
    v15 <- CallFunction (guarded) v13, [v13, v13, v12, v11]
EndIf
v16 <- LoadFloat '2.2250738585072014e-308'
v17 <- LoadString 'o'
v18 <- LoadString '9223372036854775807'
v19 <- CallMethod v18, 'includes', []
v20 <- Compare v17, '>=', v18
// Executing code generator SimpleForLoopGenerator
BeginForLoopInitializer
    v21 <- LoadInteger '0'
BeginForLoopCondition -> v22
    v23 <- LoadInteger '9'
    v24 <- Compare v22, '<', v23
BeginForLoopAfterthought v24 -> v25
    v26 <- UnaryOperation v25, '++'
BeginForLoopBody -> v27
    // Executing code generator IntArrayGenerator
    v28 <- CreateIntArray [-536870912, 60598, -36142]
    v29 <- CreateIntArray [10, 124077463, -7, -9007199254740990, 1, 208246237, 11, 2, -256]
    v30 <- CreateIntArray [-4294967297, -4257, -632517983, 771473664, 13, -74447244, -268435456]
    // Code generator finished
EndForLoop
// Code generator finished
v31 <- LoadFloat '101.4735377839952'
// Program may be interesting due to new coverage: 2680 newly discovered edges in the CFG of the target


// ===== [ Program E85D6D3A-40AC-4488-AE26-C1F00C1349B8 ] =====
// Minimizing 28AFDA08-4BBB-4ACB-96B0-11D44389B683
v0 <- LoadString 'o'
v1 <- LoadString 'undefined'
v2 <- LoadInteger '-9007199254740991'
v3 <- CallMethod (guarded) v1, 'includes', [v2, v2]
v4 <- CallFunction (guarded) v3, []
BeginRepeatLoop '5' -> v5
    v6 <- CreateIntArray [-536870912, 60598, -36142]
    v7 <- CreateIntArray [10, 124077463, -7, -9007199254740990, 1, 208246237, 11, 2, -256]
    v8 <- CreateIntArray [-4294967297, -4257, -632517983, 771473664, 13, -74447244, -268435456]
EndRepeatLoop
// Program is interesting due to new coverage: 141 newly discovered edges in the CFG of the target
// Imported program is interesting due to new coverage: 37 newly discovered edges in the CFG of the target
// Imported program is interesting due to new coverage: 154 newly discovered edges in the CFG of the target
// Imported program is interesting due to new coverage: 4 newly discovered edges in the CFG of the target
```

### CODE REVIEW

File: Sources/Fuzzilli/Engines/FuzzEngine.swift [hunk 1]
```diff
@@ -36,6 +36,9 @@ extension FuzzEngine {
                 if let aspects = fuzzer.evaluator.evaluate(execution) {
                     if fuzzer.config.enableInspection {
                         program.comments.add("Program may be interesting due to \(aspects)", at: .footer)
+                        if fuzzer.config.enableArgumentRandomization {
+                            program.comments.add("ARGS: \(fuzzer.runner.processArguments.joined(separator: " "))", at: .header)
+                        }
```
Comment (0xedward) [inline Sources/Fuzzilli/Engines/FuzzEngine.swift:41]: Let me know if there is a better place for this block :). I was also consider putting this block in `Storage.swift` between the following lines instead:

https://github.com/googleprojectzero/fuzzilli/blob/48163c833b1ac42b05058e7a73256967a1f741c3/Sources/Fuzzilli/Modules/Storage.swift#L176-L177

**Comment by saelo** [comment]
Thanks! @carl-smith can you review this one?

**Review by carl-smith — CHANGES_REQUESTED** [review]
Oh sorry Edward, I forgot to submit the comments I made last month!

**Comment by 0xedward** [comment]
> Oh sorry Edward, I forgot to submit the comments I made last month!

Ah no problem :). Thanks for the review, Carl! Sorry I only managed to find the time recently to make the changes you suggested

File: Sources/Fuzzilli/Engines/FuzzEngine.swift (hunk 1 above)
Comment (0xedward) [reply Sources/Fuzzilli/Engines/FuzzEngine.swift]: > Additionally I think we can just drop the change in Configuration.swift and always write the arguments in the .fuzzil.history, maybe "RUNNER ARGS: " to be a bit more verbose. WDYT?

Yep, that sounds good to me. I made the two changes you suggested :)

**Pull request merged**